ledger/internal/eval_test.go

```diff
@@ -382,15 +385,17 @@ func TestCowStateProof(t *testing.T) {
 		Version:           protocol.ConsensusFuture,
 	}
 
-	// if atRound header is missing the apply should fail
-	blockErr[atRound] = noBlockErr
+	// crypto verification should fail since it is not a valid stateproof
 	err = apply.StateProof(stateProofTx, atRound, c0, validate)
-	require.ErrorIs(t, err, noBlockErr)
-
-	delete(blockErr, atRound)
-	err = apply.StateProof(stateProofTx, atRound, c0, validate)
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "crypto error")
+	a.Error(err)
+	a.Contains(err.Error(), "crypto error")
+
+	a.Equal(basics.Round(512), c0.mods.StateProofNext)
+	// transaction should be applied without stateproof validation (no context, blockheader or valid stateproof needed as it represents a node catching up)
+	err = apply.StateProof(stateProofTx, atRound, c0, false)
+	a.NoError(err)
+	// make sure that the StateProofNext was updated correctly after applying
+	a.Equal(basics.Round(512+config.Consensus[protocol.ConsensusFuture].StateProofInterval), c0.mods.StateProofNext)
 }
 
 // a couple trivial tests that don't need setup
```

Comment:
Nowhere good to really put this - how about adding (or modifying) a test to check the behavior during reloadLedger at ledger_test.go?